src/integrators/sdirk_integrators.jl

```diff
@@ -42,7 +42,7 @@ end
   ndz = integrator.opts.internalnorm(dz)
   z = z .+ dz
 
-  η = max(cache.ηold,eps(first(u)))^(0.8)
+  η = max(cache.ηold,eps(first(integrator.opts.reltol)))^(0.8)
```

Comment:
why this change? I'm not sure this makes sense?

Reply:
Oh I see, nevermind.

Reply:
`u` is a number or an array of some type which might contain units, `reltol` however is a number or an array of the unitless type of elements in `u`. So `first(u)` can fail, but `first(integrator.opts.reltol)` won't (or otherwise also the calculation of residuals will fail) and it will still return the same unitless value for unitless types of `u`. Moreover, at first glance, I couldn't see another field with a unitless type of `u`.